lib/MySQL_Session.cpp

```diff
@@ -6410,22 +6410,24 @@ void MySQL_Session::handler___client_DSS_QUERY_SENT___server_DSS_NOT_INITIALIZED
 			}
 		}
 		if (session_fast_forward == false) {
-			if (qpo->gtid_from_hostgroup >= 0) {
+			if (qpo->min_gtid) {
+				gtid_uuid = qpo->min_gtid;
+			} else if (qpo->gtid_from_hostgroup >= 0) {
 				_gtid_from_backend = find_backend(qpo->gtid_from_hostgroup);
 				if (_gtid_from_backend) {
 					if (_gtid_from_backend->gtid_uuid[0]) {
-						with_gtid = true;
+						gtid_uuid = _gtid_from_backend->gtid_uuid;
 					}
 				}
 			}
-			if (with_gtid) {
-				int l = index(_gtid_from_backend->gtid_uuid,':') - _gtid_from_backend->gtid_uuid;
-				trxid = strtoull(index(_gtid_from_backend->gtid_uuid,':')+1, NULL, 10);
+			if (gtid_uuid != NULL) {
+				int l = index(gtid_uuid,':') - gtid_uuid;
```

Comment:
If the invalid value of gtid is passed into min_gtid, the proxysql crashes.
For example, the following query crashes proxysql
```
select /* min_gtid=111 */ @@hostname
```
The `char *index(const char* s, int c);` function returns a pointer to the matched character or NULL if the character is not found. The index return value should be checked for NULL.

Reply:
Event min_gtid is not correct and wrong value is used to find server, the proxysql should not crash. The client will be just timed out. And it looks like correct behavior